File: algorithm/move-zeroes/move-zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int i, j, count = 0;
    for (i = 0, j = 0; i < numsSize; i++) {
        if (nums[j] != 0) {
            j++;
            continue;
        }
        if (nums[i] != 0 && nums[j] == 0) {
            nums[j] = nums[i];
            nums[i] = 0;
            j++;
            count++;
        }
    }
}

File: algorithm/move-zeroes/move-zeroes.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
__date__= ' 9 02, 2016 '
__author__= 'mkfsn'


class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        zeros = nums.count(0)
        for i in range(zeros):
            nums.remove(0)
            nums.append(0)
